src/Microsoft.FeatureManagement.AspNetCore/AspNetCoreFeatureManagementBuilderExtensions.cs

```diff
@@ -44,5 +47,28 @@ public static IFeatureManagementBuilder UseDisabledFeaturesHandler(this IFeature
 
             return builder;
         }
+
+        /// <summary>
+        /// Adds the <see cref="DefaultHttpTargetingContextAccessor"/> to be used for targeting and registers the targeting filter to the feature management system.
+        /// </summary>
+        /// <param name="builder">The <see cref="IFeatureManagementBuilder"/> used to customize feature management functionality.</param>
+        /// <returns>A <see cref="IFeatureManagementBuilder"/> that can be used to customize feature management functionality.</returns>
+        public static IFeatureManagementBuilder WithTargeting(this IFeatureManagementBuilder builder)
```

Comment:
Should we call `AddHttpContextAccessor` in this method since our default targeting accessor depends on it.

Reply:
That's a great question. I originally thought no- so we're in parity with what we have today. However it would remove another hurdle to get started so I think it's a good idea.

Reply:
I'd like to decouple that.

By the way, is this method name the one that we discussed so many options and couldn't settle on one?

Reply:
Okay, let's keep it decoupled. Could always do that in a separate PR anyway.

> By the way, is this method name the one that we discussed so many options and couldn't settle on one?

Yes it is. The last design review it was discussed up was the caching meeting last month. We lightly aligned on `.WithTelemetry<DefaultTargetingContextAccessor>()`, but some smaller discussion afterwards lead to `.WithTelemetry()` so we didn't have to make `DefaultTargetingContextAccessor` public.

Although I'm having trouble finding a paper trail to confirm that we did commit to this.

Reply:
@zhenlan do you have thoughts on the `.WithTelemetry()` over `.WithTelemetry<DefaultTargetingContextAccessor>()`

Reply:
Do you mean WithTargeting?

Reply:
Yes, this would have been referring to `WithTargeting`, not `WithTelemetry`

Reply:
Would a blazor user take a dependency on this package and see this method? My thought is that if we go with WithTargeting, any application that would depend on this package should most likely use this method.

Reply:
My fault- yes `.WithTargeting`. 

Blazor Webassembly is just a client side thing- so it would work fine with an ASP.NET backend requiring this.

Blazor Server would not work as expected- and I wouldn't expect the dev to use the ASP.NET package. Although- Blazor Server are built on top of ASP.NET Core so it's not unreasonable a developer would try this.

Reply:
We could add it by default if 1) it will not break any existing applications and 2) users can easily override it if they want to use their own targeting context accessor.

Reply:
Okay, having it with just `WithTargeting` sounds reasonable to me.